### PR TITLE
Create spaceship cockpit

### PR DESCRIPTION
**Pull request by ItsAdel**
Did not run linter yet.
Implementation for rendering all the spaceships parts done.

### CODE REVIEW

File: templates/vue/src/components/tyde/TydeCockpit.vue [hunk 1]
```diff
@@ -1,42 +1,29 @@
...
-        v-for="log in visibleLogs"
-        :key="log.name"
-        :log="log"
-        @close="$emit('close')"
-      />
-    </ul>
+  <section id="tyde-cockpit" :style="cockpitImage">
+    <div v-for="node in this.nodes" :key="node.id">
+      <div
+        v-if="isNodeModuleType(node)"
```
Comment (narendrasss) [inline templates/vue/src/components/tyde/TydeCockpit.vue]: `v-for` with `v-if` sometimes doesn't work as expected. I would suggest adding a computed property that filters this out, e.g.

```js
modules() {
  return this.nodes.filter(isNodeModuleType)
}
```

File: templates/vue/src/components/tyde/TydeCockpit.vue [hunk 2]
```diff
@@ -65,6 +53,12 @@ export default {
       }
       return this.logsWithAnswers
     },
+    cockpitImage() {
+      return {
+        backgroundImage: `url(${this.settings.spaceshipBackgroundUrl})`
+      }
```
Comment (narendrasss) [inline templates/vue/src/components/tyde/TydeCockpit.vue]: I'm not able to change this setting anywhere in the UI. Not sure if that's intended. Can you clarify @wynnset ?

File: templates/vue/src/components/tyde/TydeSpaceship.vue [hunk 3]
```diff
@@ -55,9 +56,9 @@ export default {
   },
   computed: {
     ...mapState(["lightbox"]),
-    backpackUrl() {
-      return `${wpData.vue_uri}/${BackpackIcon.split("dist")[1]}`
-    },
+    spaceshipIconUrl() {
+      return `${wpData.vue_uri}/${SpaceshipIcon.split("dist")[1]}`
+    }
```
Comment (narendrasss) [inline templates/vue/src/components/tyde/TydeSpaceship.vue]: Please use the `getImagePath` function available in the `Helpers` class instead.

File: templates/vue/src/components/tyde/TydeCockpit.vue (hunk 2 above)
Comment (wynnset) [reply templates/vue/src/components/tyde/TydeCockpit.vue]: You can change the setting by going to the Tapestry settings (the gear icon where you also can change the background for the Tapestry or turn auto-layout on/off) and then going to the TYDE tab.

File: templates/vue/src/components/tyde/TydeCockpit.vue [hunk 4]
```diff
@@ -126,4 +147,5 @@ export default {
     overflow-y: scroll;
   }
 }
+
 </style>
```
Comment (wynnset) [inline templates/vue/src/components/tyde/TydeCockpit.vue]: Please keep the original TydeMenuHome component. We may want to use it still in the future.

File: templates/vue/src/components/tyde/TydeCockpit.vue [hunk 5]
```diff
@@ -105,12 +99,39 @@ export default {
...
 }
 </script>
 
 <style lang="scss" scoped>
-#tyde-menu-home {
+#tyde-cockpit {
+  height: 85vh;
   nav ul {
     display: flex;
     font-size: 16px;
```
Comment (wynnset) [inline templates/vue/src/components/tyde/TydeCockpit.vue]: `nav ul` and `.logs` is not used in the TydeCockpit component. Please remove.

File: templates/vue/src/components/tyde/TydeCockpit.vue [hunk 6]
```diff
@@ -105,12 +100,45 @@ export default {
...
+    },
+    isNodeModuleType(node) {
+      return node.tydeType === tydeTypes.MODULE
+    },
+    nodeMouseOverHandler(node) {
+      document.getElementById("tyde-cockpit-"+node.id).style.backgroundImage = `url(${node.typeData.spaceShipPartHoverIconUrl})`
+    },
+    nodeMouseLeaveHandler(node) {
+      document.getElementById("tyde-cockpit-"+node.id).style.backgroundImage = `url(${this.moduleImage(node)})`
+    },
```
Comment (wynnset) [inline templates/vue/src/components/tyde/TydeCockpit.vue]: This is not the right way of doing things in Vue. I recommend creating a component for a spaceship part. Inside that component, keep a state to determine which image should be used. Update this state when the user hovers over or away, and tie the background image to the state. Let me know if you want to chat about this to either myself or Narendra.

**Pull request merged**